### PR TITLE
ConfigurationManager fix for UAPAOT

### PR DESCRIPTION
**Pull request by tijoytom**
Use Assembly.CodeBase is not available on
uapaot,replacing with 
Path.Combine(AppDomain.CurrentDomain.BaseDirectory,
assembly.ManifestModule.Name)

### CODE REVIEW

**Comment by tijoytom** [comment]
@AtsushiKan

**Comment by stephentoub** [comment]
cc: @JeremyKuhne

**Comment by danmosemsft** [comment]
@tijoytom per the email thread last week, do I understand right that @MichalStrehovsky will be enabling Assembly.CodeBase to produce something reasonable? Ie., you just don't want to hold up on that.

**Comment by MichalStrehovsky** [comment]
> do I understand right that @MichalStrehovsky will be enabling Assembly.CodeBase to produce something reasonable

@AtsushiKan would be the one making that change. After a lengthy email discussion we decided that on Project N, in cases when the original managed assemblies no longer exist because they were fully nativized, `CodeBase` will return what `ManifestModule.Name` would return - so not a full path. It's what Mono is doing, and it's not usable for this scenario because of the lack of a rooted path.

File: src/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs [hunk 1]
```diff
@@ -60,15 +60,17 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 // used for local paths and "file://" for UNCs. Simply removing the prefix will make
                 // local paths relative on Unix (e.g. "file:///home" will become "home" instead of
                 // "/home").
-                Uri uri = new Uri(exeAssembly.CodeBase);
+                string configBasePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, exeAssembly.GetName().Name);
```
Comment (tijoytom) [inline src/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs]: OK , I will use ManifestModule.Name then

**Comment by tijoytom** [comment]
restest Innerloop netfx Windows_NT Debug x86 Build and Test

**Comment by tijoytom** [comment]
retest Portable Linux x64 Debug Build

**Comment by tijoytom** [comment]
Portable Linux x64 Debug Build  failure seems to be infrastructure

**Comment by tijoytom** [comment]
@JeremyKuhne Is this good for merge?

**Comment by JeremyKuhne** [comment]
Sorry, I was OOF. Looks good.

**Pull request merged**